### PR TITLE
Overhaul documentation

### LINKED ISSUE #18: Document migrations commands
- [x] Document `dotnet ef migrations add`
- [x] Document `dotnet ef migrations remove`
- [x] Document `dotnet ef migrations script ...`

### PR DESCRIPTION
**Pull request by hburn7**
Closes #18 

- Restructures API setup docs
- Breaks some sections into multiple pages
- Reorganizes content layout (i.e. database backup & recovery should not be under the API section)
- Adds contribution guide

### CODE REVIEW

File: docs/topics/o-TR-Docs.md [hunk 1]
```diff
@@ -2,4 +2,8 @@
 
 The o!TR Docs serve to document anything and everything related to the o!TR project. These docs are written using [Jetbrains Writerside](https://www.jetbrains.com/writerside/).
```
Comment (cytusine0) [inline docs/topics/o-TR-Docs.md:3]: capitalize JetBrains? (also in line 75 of `API-Getting-Started.md`)

File: docs/topics/Migrations.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
+# Migrations
+
+Migrations are handled by [Entity Framework](https://learn.microsoft.com/en-us/aspnet/entity-framework). These are auto-generated files and configurations which enable code-first changes to the database.
+
+## Creating / Editing Entities
+
+Migrations must be created and tested when any of the following is modified:
```
Comment (cytusine0) [inline docs/topics/Migrations.md:7]:
```suggestion
Migrations must be created and tested when any of the following are modified:
```

File: docs/topics/Migrations.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+1. Files in `Database/Entities`. These objects directly link to a database table.
+    - Only create a migration when the changes modify the entity such that a database change is required. For example, adding a new field requires a migration, but modifying basic comments do not.
```
Comment (cytusine0) [inline docs/topics/Migrations.md:10]:
```suggestion
    - Only create a migration when the changes modify the entity such that a database change is required. For example, adding a new field requires a migration, but modifying basic comments does not.
```

File: docs/topics/Migrations.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+To create a new migration, run the following in the root directory of the repository.
+
+```Shell
+dotnet ef migrations add [Entity_ChangesMade] \
+--project Database \
+--startup-project API \
+--context OtrContext
+```
+
+> `[Entity_ChangesMade]` is the name of your migration. Migrations should use PascalCase with underscores separating the entity from the changes made. For example, `Player_ConvertIdToInteger` is a good name for a migration as it describes the change made at a glance.
```
Comment (cytusine0) [inline docs/topics/Migrations.md:24]:
```suggestion
> `[Entity_ChangesMade]` should be replaced with the name of your migration. Migrations should use PascalCase with underscores separating the entity from the changes made. For example, `Player_ConvertIdToInteger` is a good name for a migration as it describes the change made at a glance.
```

File: docs/topics/Development.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# Development
+
+Start typing here...
```
Comment (cytusine0) [inline docs/topics/Development.md:3]: do you want a table of contents here too?

File: docs/topics/Contribution-Guide.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
+# Contribution Guide
+
+Those who wish to make open-source contributions to the API must adhere to this contribution guide.
+
+## Claiming an Issue
+
+First-time contributors may volunteer to take low-priority tasks to work on. We request first-time contributors take low-priority issues due to the length of time it takes a first-time developer to contribute to the project.
```
Comment (cytusine0) [inline docs/topics/Contribution-Guide.md:7]: This seems a bit wordy, is the reasoning here about difficulty of onboarding, or time needed for code review, or something else?

File: docs/topics/o-TR-Docs.md (hunk 1 above)
Comment (cytusine0) [reply docs/topics/o-TR-Docs.md:3]: also line 5 of ``Getting-Started.md``if consistency matters

separately, line 11 of ``Getting-Started.md`` says "VS Code" instead of "Visual Studio Code" 🤓

File: docs/topics/API-Configuration.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+      Set the fields
+      <tip>This callback URL is used for otr-web setup</tip>
+      <img src="new-oauth-app-fields.png" alt="New OAuth app fields"/>
+   </step>
+   <step>
+      Sample fields <img src="sample-oauth-app.png" alt="Sample OAuth app with fields"/>
+   </step>
+</procedure>
+
+### Jwt
```
Comment (cytusine0) [inline docs/topics/API-Configuration.md:64]: capitalized? i guess?

**Review by cytusine0 — COMMENTED** [review]
I don't know what most of this means but I did scan through 👍

File: docs/topics/Migrations.md (hunk 3 above)
Comment (hburn7) [reply docs/topics/Migrations.md:10]: image

File: docs/topics/API-Configuration.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+Configure the o!TR API with configuration files.
+
+## Configuration file
+
+Duplicate the `example.appsettings.json` file and rename to `appsettings.Development.json`. Configure the fields as described below.
+
+> Ensure the file is located in the `API` project folder (the same folder which contains `API.csproj`).
+>
+
+## Fields
```
Comment (myssto) [inline docs/topics/API-Configuration.md:12]:
```suggestion
## Configuration Sections
```

**Review by myssto — CHANGES_REQUESTED** [review]
Strictly on the information side, not much of anything to complain about it looks solid

**Comment by hburn7** [comment]
Made changes as requested.
